builders/build/canonical-contracts/precompiles/batch.md

```diff
@@ -0,0 +1,225 @@
+
+Try finding a transaction's call data using Remix:
+
+1. Expand the SimpleMessage contract under **Deployed Contracts**
+2. Expand the **`setMessage`** function
+3. Enter a number and a message of your choice for **id** & **message**. In this example, **id** will be 1 and **message** will be "moonbeam"
+4. Instead of sending the transaction, click the copy button next to the **transact** button to copy the call data
+
+![Transaction Call Data](/images/builders/build/canonical-contracts/precompiles/batch/batch-5.png)
+
+Now you have the transaction's call data! Considering the example values of 1 and "moonbeam", we can keep an eye out for their encoded values in the call data:
+
+ --8<-- 'text/batch/simple-message-call-data.md'
+
+The calldata can be broken into 5 lines, where the first line is the function selector and the next four are the input data. 
+
+Note that the second line is equal to 1, which is the **id** that was provided. 
+
+What's left has to do with the **message** input. These last three lines are tricky, since strings are encoded with [RLP encoding](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/){target=_blank}. For sake of simplicity, understand that the third line refers to an offset, the fourth line refers to the string's length, and the fifth line is "moonbeam" in hexadecimal format with zeros for padding.
+
+### Function Interaction via Precompile {: #function-interaction-via-precompile }
+
+Interacting with a function is very similar to [sending a native currency](#send-native-currency-via-precompile), since they are both transactions. However, call data is required to properly provide input to functions and a sender may desire to limit the amount of gas spent in each subtransaction. 
+
+The *call_data* and *gas_limit* transactions more relevant for subtransactions that interact with contracts. For each function in the Batch interface, the *call_data* input is an array that corresponds to the call data for each subtransaction. If its length is less than the *to* input, the remaining subtransactions will have no call data. The *gas_limit* input is an array that corresponds to the amount of gas that each can spend for each subtransaction. If its length is less than the *to* input, the remaining transactions will have all of the batch transaction's remaining gas forwarded.
+
+Try using the precompile to send a transaction:
+
+1. Copy the SimpleMessage contract's address with the copy button on the right side of its header. Be sure to also have the [call data from the previous section](#finding-a-contract-interactions-call-data) 
+2. Expand the Batch contract under **Deployed Contracts**
+3. Expand the **batchAll** function
```

Comment:
Mentione somewhere (here or before) that for this example you'll be using the batchAll that ensures transactions are resolved atomically, and mention the other options